lib/src/front_end/comment_writer.dart

```diff
@@ -106,6 +123,14 @@ class CommentWriter {
         if (comment == token.precedingComments) linesBefore = 2;
       }
 
+      // If this comment configures the page width, remember it.
+      if (_widthCommentPattern.firstMatch(text) case var match?
+          when _pageWidthFromComment == null) {
```

Comment:
[optional] Consider nesting the conditionals with the null check first to avoid doing regex work when it won't be used.